File: backend/core/responses/baseStationResponse.py
from typing import Literal

from pydantic import BaseModel


class BaseStationResponse(BaseModel):
    """Base response model for a station information."""

    type: Literal["station"] = "station"
    provider: str
    publicId: str
    providerUrl: str
    name: str
    imageUrl: str
    streamUrl: str | None = None
    country: str | None = None
    codec: str | None = None
    bitrate: int | None = None
    tags: str | None = None
    homepage: str | None = None

File: frontend/packages/dto/baseStationResponse.ts
import { z } from "zod";

export const BaseStationResponseSchema = z.object({
    type: z.union([z.literal("station")]).default("station"),
    provider: z.string(),
    publicId: z.string(),
    url: z.string(),
    name: z.string(),
    imageUrl: z.string(),
});

export type BaseStationResponse = z.infer<typeof BaseStationResponseSchema>;
